ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -523,6 +524,15 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
       final StatsUpdater su = computeStats ? StatsUpdater.init(ci, msc.findColumnsWithStats(
           CompactionInfo.compactionInfoToStruct(ci)), conf,
           runJobAsSelf(ci.runAs) ? ci.runAs : t.getOwner()) : null;
+      // result directory for compactor to write new files
+      Path resultDir = null;
+      if (ci.type == CompactionType.MAJOR) {
```

Comment:
Could this be simplified to just: 
Path resultDir = QueryCompactor.Util.getCompactionResultDir(sd, tblValidWriteIds, conf, 
             ci.type == CompactionType.MAJOR, false, false, null);